### PR TITLE
22391 - EFT Multi Account under payment

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/22391

*Description of changes:*
- Add statement summary multi account under payment logic
- short name service refactor for eft credit balance to avoid cyclic issue

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 1]
```diff
@@ -75,16 +75,6 @@ class EFTShortnamesSearch:  # pylint: disable=too-many-instance-attributes
 class EFTShortnames:  # pylint: disable=too-many-instance-attributes
     """Service to manage EFT short name model operations."""
 
-    @staticmethod
-    def get_eft_credit_balance(short_name_id: int) -> Decimal:
```
Comment (ochiu) [inline pay-api/src/pay_api/services/eft_short_names.py:79]: Moved to model due to cyclic issues with statement service.

As previously discussed we will be refactoring this service to split it out later. Depending on the query some of these functions are probably better in the model.

File: pay-api/src/pay_api/services/statement.py [hunk 2]
```diff
@@ -392,9 +393,46 @@ def get_summary(auth_account_id: str, statement_id: str = None):
...
+        invoices_subquery = (db.session.query(distinct(InvoiceModel.id),
+                                              InvoiceModel.id,
+                                              InvoiceModel.total,
+                                              InvoiceModel.paid)
+                             .join(StatementInvoicesModel, StatementInvoicesModel.invoice_id == InvoiceModel.id)
+                             .filter(and_(InvoiceModel.payment_method_code == PaymentMethod.EFT.value,
+                                          InvoiceModel.invoice_status_code.in_([InvoiceStatus.APPROVED.value,
+                                                                                InvoiceStatus.OVERDUE.value])))
+                             .filter(
+            InvoiceModel.payment_account_id.in_(accounts_query)).subquery())
```
Comment (ochiu) [inline pay-api/src/pay_api/services/statement.py:422]: Tested against several large accounts for this in clause, at once. Results came back in about 100ms

**Comment by seeker25** [comment]
Hold off for release

**Comment by ochiu** [comment]
Working on a test to set up multi account link to trigger under payment flag.

File: pay-api/src/pay_api/models/eft_short_name_links.py [hunk 3]
```diff
@@ -63,25 +63,35 @@ class EFTShortnameLinks(Versioned, BaseModel):  # pylint: disable=too-many-insta
     updated_by_name = db.Column('updated_by_name', db.String(100), nullable=True)
     updated_on = db.Column('updated_on', db.DateTime, nullable=True)
 
+    active_statuses = [EFTShortnameStatus.LINKED.value,
+                       EFTShortnameStatus.PENDING.value]
+
     @classmethod
-    def find_by_short_name_id(cls, short_name_id: int):
+    def find_by_short_name_id(cls, short_name_id: int) -> Self:
```
Comment (seeker25) [inline pay-api/src/pay_api/models/eft_short_name_links.py:70]: cool

File: pay-api/src/pay_api/services/statement.py [hunk 4]
```diff
@@ -392,9 +393,46 @@ def get_summary(auth_account_id: str, statement_id: str = None):
             'total_invoice_due': float(invoices_unpaid_amount) if invoices_unpaid_amount else 0,
             'total_due': total_due,
             'oldest_due_date': oldest_due_date,
-            'short_name_links_count': short_name_links_count
+            'short_name_links_count': short_name_links_count,
+            'is_eft_under_payment': Statement._is_eft_under_payment(auth_account_id, total_due)
+            if calculate_under_payment else None
```
[the remark this answers is not in the document]
Comment (seeker25) [reply pay-api/src/pay_api/services/statement.py:398]: param as part of the function call?

**Review by seeker25 — APPROVED** [review]

**Pull request merged**